components/rich_text_editor/extensions/link/utils.js

```diff
@@ -0,0 +1,169 @@
+import { getMarksBetween } from '@tiptap/core';
```

Comment:
These are utils for the autolink plugin and contain the bulk of the logic. I wrote them in a semi-generic fashion, but since they're pretty specific to the autolink functionality I didn't want to go overboard and make everything into a fully generic util. We can pull some out into the common util file if they seem useful for other purposes and/or perhaps in the future we'll want to build similar plugins that could share some of the utils, but for now I don't know about such needs.